05-UI-ReactJS/source/js/components/views/movie-form.js

```diff
@@ -0,0 +1,88 @@
+import React, { PropTypes } from "react";
+import Movie from "../../movie.js";
+import store from "../../store";
+import { browserHistory } from 'react-router';
+
+export default class MovieForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.state = {
+      movie: new Movie()
+    };
+  }
+
+  render() {
+    let mode;
+    this.state.movie.id ? mode = "editing" : mode = "creating";
```

Comment:
This could be refactor to 
`let mode = (this.state.movie.id) ? "editing" : "creating";`

Also remember that is a best practice to wrap the condition between parenthesis.